RFM69 radio link integration

muebau opened the pull request.

lakeman commented:
It looks like you're exclusively using TAB characters for indentation. We've been using the convention of indenting by 2 spaces, with TAB's representing 8 spaces. This difference makes it hard to see what you have actually changed.

Also it looks like you could easily rebase your last commit onto our development branch without needing to add any unnecessary merges into our commit history.

muebau commented:
Hello,
what I would do:
1. create a new (temp) branch from development.
2. cherry pick my changes
3. fix the formatting
4. reset (--hard) the development branch to the (temp) branch
5. send another pull request

For the future I would do "git pull --rebase" to avoid the problem in the future.

Would this help?
Do you have any suggestions of improvement?

muebau committed:
initial commit to integrate a new radio link (RFM69)

lakeman commented:
git pull --rebase
and
git rebase -i [remote name]/development

Just create new commits until you think the current version is right, then
use rebase -i to squash everything that you think should be combined into
one patch.
While the end result is the same as reset --hard and / or cherry-picking,
rebase -i is a bit easier to control without losing changes IMHO.

On Tue, Oct 7, 2014 at 5:43 PM, muebau notifications@github.com wrote:

> Hello,
> what I would do:
> 1. create a new (temp) branch from development.
> 2. cherry pick my changes
> 3. fix the formatting
> 4. reset (--hard) the development branch to the (temp) branch
> 5. send another pull request
> 
> For the future I would do "git pull --rebase" to avoid the problem in the
> future.
> 
> Would this help?
> Do you have any suggestions of improvement?
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/servalproject/serval-dna/pull/87#issuecomment-58145666
> .

muebau committed:
state mashine ready, hopefully the first running version

muebau committed:
first running version. nothing transmitted yet.

muebau committed:
some little improvements.

Nothing transmitted yet. Do we miss a heartbeat?

muebau committed:
propper formatted

muebau committed:
IN/OUT/RETURN fixed

muebau committed:
fixed offsets

lakeman committed:
Pass PIE flags for compiling on Android L

muebau committed:
a commit to save the current state

muebau committed:
add header files to make file definition

muebau committed:
corrected header file

muebau committed:
current serval.conf

muebau committed:
a little cleanup to find the bug

muebau committed:
working version. packets go from a to b.

muebau committed:
little improvements

muebau committed:
before system upgrade

muebau committed:
Looks like the radio works. Now need some traffic to test it under heavy load.

muebau committed:
its running with many glitches

muebau committed:
simplyfied RX timeout

muebau committed:
try a simpler (synchron) approach

muebau committed:
straightened the state machine

muebau committed:
little fix, more debug messages

lakeman committed:
Fix minor compilation issues on CentOS.

- ensure size_t is defined when compiling xprintf.h
- declare overlay_route_record as a union field in overlay_route_record to avoid alignment issues
- set the result of fwrite and fputs
- support slightly older git version

muebau committed:
radio works now

It could made an improvement to avoid collitions. Unfortunately the
radios start to transmit all after the where interupted be a
transmitting node.
